File: JS/Contagem.js
function contarInteirosNoIntervalo(dados, n) {
    if (!Array.isArray(dados) || dados.length === 0) {
        return 0;
    }

    const primeiroDado = dados[0];
    const limiteInferior = Math.min(primeiroDado, n);
    const limiteSuperior = Math.max(primeiroDado, n);

    return dados.filter((valor) =>
        Number.isInteger(valor) &&
        valor >= limiteInferior &&
        valor <= limiteSuperior
    ).length;
}

const conjunto = [3, 5, 7, 8, 10, 12, 15];
const n = 15;

console.log(`Valores inteiros entre ${conjunto[0]} e ${n}: ${contarInteirosNoIntervalo(conjunto, n)}`);

File: Java/algoritmos/Contagem.java
import java.util.List;

public class Contagem {
    public int contarInteirosNoIntervalo(List<Integer> dados, int n) {
        if (dados == null || dados.isEmpty()) {
            return 0;
        }

        int primeiroDado = dados.get(0);
        int limiteInferior = Math.min(primeiroDado, n);
        int limiteSuperior = Math.max(primeiroDado, n);
        int contador = 0;

        for (int valor : dados) {
            if (valor >= limiteInferior && valor <= limiteSuperior) {
                contador++;
            }
        }

        return contador;
    }
}
